lib/saluki-components/src/destinations/datadog_metrics/mod.rs

```diff
@@ -383,6 +407,7 @@ async fn run_io_loop<O>(
                 }
             }
             Err(e) => {
+                let e: tower::BoxError = e.into();
```

Comment:
Does it not actually work without the `into` call? `e` should already be `Box<dyn Error + ...>`.

Reply:
It does not. Compiler shows `e` as `<S as Service<Request<ChunkedBuffer<O>>>>::Error` when I hover over it.

Reply:
Oh, right, duh.

Change line 378 to be `S::Error:: std::error::Error + Send` and then you can drop the `into` down here.

Reply:
For posterity: we paired on this because doing my suggested change still results in the compiler getting very angry about `DatadogMetrics<O, S>` being unsized and not being able to box it and yadda yadda.

Not worth fighting with the compiler over this, but maybe something to try and poke at on a rainy day. :shrug: